src/include/miopen/solver.hpp

```diff
@@ -206,19 +250,19 @@ struct PerformanceConfigConvAsm3x3U : Serializable<PerformanceConfigConvAsm3x3U>
     std::string ToString() const;
 };
 
-struct ConvAsm3x3U : ConvSolver
+struct ConvAsm3x3U : ConvTunableSolver<PerformanceConfigConvAsm3x3U>
 {
     const std::string& SolverDbId() const override { return GetSolverDbId<ConvAsm3x3U>(); }
 
     bool IsApplicable(const ConvolutionContext& params) const override;
-    PerformanceConfigConvAsm3x3U GetPerformanceConfig(const ConvolutionContext&) const;
+
+    PerformanceConfigConvAsm3x3U GetDefaultPerformanceConfig(const ConvolutionContext&) const final;
     bool IsValidPerformanceConfig(const ConvolutionContext&,
-                                  const PerformanceConfigConvAsm3x3U&) const;
+                                  const PerformanceConfigConvAsm3x3U&) const final;
```

Comment:
`final` here disallows overriding in descendants. Why we want this? I guess this is because this class is not intended to be inherited from, right? But in the such a case `final` should be applied to a class definition.

Reply:
Tunable solvers have been marked as `final`. Thanks!

Reply:
resolved